### PR TITLE
[LogicApps] : Integration account cmdlet bug fixes and SDK and rest-api-spec fixes. 

### PR DESCRIPTION
**Pull request by pankajsn**
## Description

This PR includes:
* Resulting changes due to bug fixes in Swagger Spec.
* Bug Fixes in PowerShell cmdlets.
* Parameter Name change in Integration account cmdlets i.e. Parameter "Name" changed to "IntegrationAccountName".
* Page size issue fixed for integration account get/list cmdlets 
---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

File: src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1 [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '2.5.0'
+ModuleVersion = '3.0.0'
```
Comment (shahabhijeet) [inline src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1]: @pankajsn we will take care of handling versions in PS modules. So can you please revert this change.

File: src/ResourceManager/LogicApp/Commands.LogicApp/Cmdlets/IntegrationAccount/GetAzureIntegrationAccountAgreementCommand.cs [hunk 2]
```diff
@@ -32,9 +32,8 @@ public class GetAzureIntegrationAccountAgreementCommand : LogicAppBaseCmdlet
         public string ResourceGroupName { get; set; }
 
         [Parameter(Mandatory = false, HelpMessage = "The integration account name.")]
-        [Alias("ResourceName")]
         [ValidateNotNullOrEmpty]
-        public string Name { get; set; }
+        public string IntegrationAccountName { get; set; }
```
Comment (shahabhijeet) [inline src/ResourceManager/LogicApp/Commands.LogicApp/Cmdlets/IntegrationAccount/GetAzureIntegrationAccountAgreementCommand.cs]: @pankajsn this is breaking change. You cannot do this. You will first have to add a warning. And when we have a breaking change release at that time you will be able to make this change. This cannot happen in this release.

File: src/ResourceManager/LogicApp/Commands.LogicApp/Cmdlets/IntegrationAccount/GetAzureIntegrationAccountCallbackUrlCommand.cs [hunk 3]
```diff
@@ -36,9 +36,8 @@ public class GetAzureIntegrationAccountCallbackUrlCommand : LogicAppBaseCmdlet
 
         [Parameter(Mandatory = true, HelpMessage = "The integration account name.",
             ValueFromPipelineByPropertyName = true)]
-        [Alias("ResourceName")]
         [ValidateNotNullOrEmpty]
-        public string Name { get; set; }
+        public string IntegrationAccountName { get; set; }
```
Comment (shahabhijeet) [inline src/ResourceManager/LogicApp/Commands.LogicApp/Cmdlets/IntegrationAccount/GetAzureIntegrationAccountCallbackUrlCommand.cs]: @pankajsn breaking change, revert this and add a warning that this will be change in future release.
Also update the upcoming breaking change file to document this breaking change coming up in future release.

File: src/ResourceManager/LogicApp/Commands.LogicApp/Cmdlets/IntegrationAccount/GetAzureIntegrationAccountCertificateCommand.cs [hunk 4]
```diff
@@ -32,9 +32,8 @@ public class GetAzureIntegrationAccountCertificateCommand : LogicAppBaseCmdlet
         public string ResourceGroupName { get; set; }
 
         [Parameter(Mandatory = false, HelpMessage = "The integration account name.")]
-        [Alias("ResourceName")]
         [ValidateNotNullOrEmpty]
-        public string Name { get; set; }
+        public string IntegrationAccountName { get; set; }
```
Comment (shahabhijeet) [inline src/ResourceManager/LogicApp/Commands.LogicApp/Cmdlets/IntegrationAccount/GetAzureIntegrationAccountCertificateCommand.cs]: @pankajsn breaking change. Revert this.

File: src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1 (hunk 1 above)
Comment (pankajsn) [inline src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1]: Version fixed

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by shahabhijeet** [comment]
started on-demand
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1410/
Once this passes we are good to merge this PR

**Comment by shahabhijeet** [comment]
on-demand succeeded. Looks good to be merged.

**Comment by pankajsn** [comment]
Thanks for update. Please merge.

**Comment by pankajsn** [comment]
@Abhijeet Shah<mailto:Abhijeet.Shah@microsoft.com>
Just following up on the below PR. I am still seeing this pull request open with [Do not merge]  & [Change Requested] labels. I hope there is nothing pending from our side and these are stale labels only.

From: Pankaj Singh Negi
Sent: Friday, February 17, 2017 3:01 PM
To: 'Azure/azure-powershell' <reply@reply.github.com>; Azure/azure-powershell <azure-powershell@noreply.github.com>
Cc: Mention <mention@noreply.github.com>; Vinay Singh <vinaysin@microsoft.com>
Subject: RE: [Azure/azure-powershell] [LogicApps] : Integration account cmdlet bug fixes and SDK and rest-api-spec fixes. (#3517)

Thanks for update. Please merge.

**Comment by pankajsn** [comment]
@Abhijeet Shah<mailto:Abhijeet.Shah@microsoft.com>
LogicApp PR is not merged in the current release. Till the last communication, we had an impression that our changes would land in the current release but it did not happen. Did we miss anything from our side ?

From: Pankaj Singh Negi
Sent: Tuesday, February 21, 2017 10:56 AM
To: 'Azure/azure-powershell' <reply@reply.github.com>; 'Azure/azure-powershell' <azure-powershell@noreply.github.com>; Abhijeet Shah <Abhijeet.Shah@microsoft.com>
Cc: 'Mention' <mention@noreply.github.com>; Vinay Singh <vinaysin@microsoft.com>
Subject: RE: [Azure/azure-powershell] [LogicApps] : Integration account cmdlet bug fixes and SDK and rest-api-spec fixes. (#3517)

@Abhijeet Shah<mailto:Abhijeet.Shah@microsoft.com>
Just following up on the below PR. I am still seeing this pull request open with [Do not merge]  & [Change Requested] labels. I hope there is nothing pending from our side and these are stale labels only.

From: Pankaj Singh Negi
Sent: Friday, February 17, 2017 3:01 PM
To: 'Azure/azure-powershell' <reply@reply.github.com>; Azure/azure-powershell <azure-powershell@noreply.github.com>
Cc: Mention <mention@noreply.github.com>; Vinay Singh <vinaysin@microsoft.com>
Subject: RE: [Azure/azure-powershell] [LogicApps] : Integration account cmdlet bug fixes and SDK and rest-api-spec fixes. (#3517)

Thanks for update. Please merge.
